onnx/external_data_helper.py

```diff
@@ -298,12 +308,21 @@ def write_external_data_tensors(model: ModelProto, filepath: str) -> ModelProto:
     Returns:
         ModelProto: The modified model object.
     """
+    file_cleared = False
+
     for tensor in _get_all_tensors(model):
         # Writing to external data happens in 2 passes:
         # 1. Tensors with raw data which pass the necessary conditions (size threshold etc) are marked for serialization
         # 2. The raw data in these tensors is serialized to a file
         # Thus serialize only if tensor has raw data and it was marked for serialization
         if uses_external_data(tensor) and tensor.HasField("raw_data"):
+            if not file_cleared:
+                # Clear the file if it exists, then open for reading and writing
+                # Clear needed for old data, to ensure file_size grows and padding is added
+                external_data_file_path = os.path.join(filepath, ExternalDataInfo(tensor).location)
+                with open(external_data_file_path, "wb"):
+                    pass
+                file_cleared = True
```

Comment:
(a) Seems to me this may not work as intended when we write each tensor to a separate file?
(b) Also, I guess this is an orthogonal improvement (trying to handle pre-existing files), independent of the alignment extension?

Reply:
It's not orthogonal in that I hit a bug when writing out the padding (zero's). It didn't work when there was a pre-existing file because the code uses the fact that the file size is increasing to know when to write zero's.

Reply:
How would it work for multiple files? If two of the files are pre-existing, file_cleared would be set to true after the first file, and so the clear logic will not be invoked for the second file.